packages/terra-cli/index.js

```diff
@@ -0,0 +1,119 @@
+const yargs = require('yargs/yargs');
+const path = require('path');
+const fs = require('fs-extra');
+
+const Logger = require('./lib/utils/Logger');
+
+const logger = new Logger({ prefix: '[terra-cli:terra]' });
+
+const DEPENDENCY_ALLOW_LIST = ['@cerner/terra-functional-testing', '@cerner/terra-open-source-scripts'];
+
+/**
+ * Does the initial set up of the CLI. This sets up things like help, version, etc.
+ */
+const setupCLI = () => {
+  const cli = yargs();
+
+  return cli.usage('Usage: $0 <command> [options]')
+    .demandCommand(1, 'A command is required. Pass --help to see all available commands and options.')
+    .recommendCommands()
+    .strict()
+    .fail((msg, err) => {
+      // certain yargs validations throw strings
+      const actual = err || new Error(msg);
+
+      // ValidationErrors are already logged, as are package errors
+      if (actual.name !== 'ValidationError' && !actual.pkg) {
+        // the recommendCommands() message is too terse
+        if (/Did you mean/.test(actual.message)) {
+          // eslint-disable-next-line no-console
+          logger.error(`Unknown command "${cli.parsed.argv._[0]}"`);
+        }
+
+        // eslint-disable-next-line no-console
+        logger.error(actual.message);
+      }
+
+      // exit non-zero so the CLI can be usefully chained
+      cli.exit(actual.code > 0 ? actual.code : 1, actual);
+    })
+    .alias('h', 'help')
+    .alias('v', 'version')
+    .wrap(cli.terminalWidth());
+};
+
+/**
+ * Takes a directory path and finds all immediate sub directories
+ * @param {string} directoryPath - the directory path for which to find sub directories
+ * @returns {Array} an array of the sub directories appended to the passed in directory path
+ */
+const subDirectoriesOfDirectory = async (directoryPath) => {
+  // Gets contents of current directory
+  const currentDirectoryEnries = await fs.readdir(directoryPath, { withFileTypes: true });
+  // Filters down to just the directories and then appends the directory names to the initial directory path
+  return currentDirectoryEnries.filter(directoryEntry => directoryEntry.isDirectory())
+    .map(directoryEntry => path.join(directoryPath, directoryEntry.name));
+};
+
+/**
+ * Takes an array of paths, appends either src or lib depending on preferSrc, appends terra-cli, and then returns an array of the the resulting directory paths that exist
+ * @param {Array} paths - the array of paths to act upon
+ * @param {boolean} preferSrc - indicates whether or not src should be used instead of lib. This should be used in cases where the package being searched is local
+ * @returns {Array} an array of paths resolved out to the terra-cli directory
+ */
+const terraClIDirectoriesFromPaths = async (paths, preferSrc) => {
```

Comment:
can we update this to an object param? it'll clear up what each param is for, especially the boolean
```suggestion
const terraClIDirectoriesFromPaths = async ({ paths, preferSrc }) => {
```

Reply:
Updated: https://github.com/cerner/terra-toolkit/pull/460/commits/5314365d74c3ef7d44104a23b5d3143723176f6d